File: Advanced Methods of Programming/Laborator1JAVA/src/main/java/service/ServiceProba.java
package service;

import domain.Proba;
import repo.RepoException;
import repo.RepoProba;

public class ServiceProba {
    private RepoProba repoProba;

    public ServiceProba(RepoProba repoProba){
        this.repoProba=repoProba;
    }

    public void save(Proba object){
        Proba proba=this.repoProba.findOne(object.getId());
        if(proba==null)
            this.repoProba.save(object);
        else throw new RepoException("Proba exista deja!");
    }

    public int size(){ return this.repoProba.size(); }

    public Proba findOne(Integer integer){
        Proba proba=this.repoProba.findOne(integer);
        if(proba==null)
            throw new RepoException("Proba nu exista!");
        else return proba;
    }

    public Iterable<Proba> findAll(){ return this.repoProba.findAll(); }

    public Integer numarInscrieri(Integer id){ return this.repoProba.numarInscrieri(id); }

}

File: Advanced Methods of Programming/Lab1C-MPP/Lab1C-MPP/Controller/ServiceProba.cs
using Lab1C_MPP.Domain;
using Lab1C_MPP.Repo;
using System.Collections.Generic;

namespace Lab1C_MPP.Controller
{
    class ServiceProba
    {
        private RepoProba repoProba;

        public ServiceProba(RepoProba repoProba)
        {
            this.repoProba = repoProba;
        }

        public void Save(Proba entity)
        {
            Proba proba = this.repoProba.FindOne(entity.GetId());
            if (proba == null)
                this.repoProba.Save(entity);
            else throw new RepositoryException("Proba exista deja!");
        }

        public int Size() { return this.repoProba.Size(); }

        public Proba FindOne(int integer)
        {
            Proba proba = this.repoProba.FindOne(integer);
            if (proba == null)
                throw new RepositoryException("Proba nu exista!");
            else return proba;
        }

        public IEnumerable<Proba> FindAll() { return this.repoProba.FindAll(); }

        public int numarInscrieri(int id) { return this.repoProba.NumarInscrieri(id); }
    }
}
